ISDR Fixes to #73 #86 #87

Rajia opened the pull request:
@stowellm  Please review. As of  right now the app is in fully working condition! It has a new splash screen consistent with that of Motion, a new icon and all known bugs/enhancements have been fixed/made.

Rajia committed:
Added Splash Screen

Rajia committed:
Modified app to use navigation controllers, created account button for loggin and added navigation bar to project view controller in storyboard

Rajia committed:
Added credential manager to viewControler class and organized code sturcture by creating folders, still need to make login actually work

Rajia committed:
Added new app icon.

Rajia committed:
Changed dimensions of app icon

Rajia committed:
Added button for changing dataset name.

Rajia committed:
Added Labels to the UI and updated the the dataset Name and project to be displayed once changed.

Rajia committed:
"fetching and merging"Merge remote-tracking branch 'upstream/master' into ra17

Rajia committed:
App conda tutorial

Rajia committed:
Changed App Icon to Approved One

Rajia committed:
Add storyboard in ISDR for tutorials

Rajia committed:
Added Tutorials

Rajia committed:
Changed Allignement for label and image

Rajia committed:
"merging upstream and master"Merge remote-tracking branch 'upstream/master' into ra17

Rajia committed:
Fixed issue with login name not actually being changed.

Rajia committed:
Fixed bug with login not actually logging in

Rajia committed:
Temporary fix for bug with browse search bar not closing

Rajia committed:
Updated tutorial instructions and images, fixed project bug

Rajia committed:
Removed Exit Button

Rajia committed:
Added Orientation lock per Mikes request

Rajia committed:
Fixing git problem

Rajia committed:
Removed labels and created them into buttons

Rajia committed:
Fixed bug with entering project number NOT uploading button

Rajia committed:
Fixed allignment, fixed bug with project being invalid before rollb being clicked, updated app icon

Rajia committed:
Fixed bug with browse project button

Rajia committed:
Modified splash screen to match that of Motions

Rajia committed:
Removed unnecessary code.

stowellm commented:
Going to reference the issues by number here so Github links to them:
#73  #86  #87

stowellm commented on Apps/DiceRoller/DiceRoller/Constants.h:
```diff
@@ -11,5 +11,15 @@
+
+#define kNAME_DIALOG_TAG 600
+
+#define kDEFAULT_DATA_SET_NAME @"My Data Set"
```
Should add a comment above this constant, e.g.
`// Data set constants`

stowellm commented on Apps/DiceRoller/DiceRoller/DiceDataController.h:
```diff
@@ -12,8 +12,8 @@
--(int)getDieNumber;
--(void) uploadDatadie1:(int)num1 die2: (int)num2 sumOfDies: (int)sumNum numOfTests: (int)numTest;
+//-(int)getDieNumber;
+//-(void) uploadDatadie1:(int)num1 die2: (int)num2 sumOfDies: (int)sumNum numOfTests: (int)numTest;
```
If this commented code is not needed, would you mind removing it?

stowellm commented on Apps/DiceRoller/DiceRoller/DiceDataController.m:
```diff
@@ -27,108 +27,5 @@ - (void)drawRect:(CGRect)rect
 }
 */
```
Same commented code idea here.  If you don't need this commented out `drawRect:(CGRect)rect` function in `DiceDataController.m`, please remove it.

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.h:
```diff
@@ -12,12 +12,12 @@
 #import "DataManager.h"
 #import "Constants.h"
 #import "Waffle.h"
+#import "ViewController.h"
```
I don't think we need `ViewController.h` imported here.  Try removing it and ensuring the app still works.

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.h:
```diff
@@ -32,6 +32,7 @@
 - (IBAction)closeBtnOnClick:(id)sender;
 
+
 -(bool)projectHasValidFields;
```
Two spaces between `- (IBAction)closeBtnOnClick:(id)sender;` and ` -(bool)projectHasValidFields;` right now - let's make it one.

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.m:
```diff
@@ -31,27 +31,24 @@ - (void)viewWillAppear:(BOOL)animated {
     if (projectValid) {
         
         int curProjID = [dm getProjectID];
         NSString *curProjIDStr = (curProjID > 100) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
```
Why is this checking if `curProjID > 100`?  Shouldn't be `> 0` instead?

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.m:
```diff
@@ -31,27 +31,24 @@ - (void)viewWillAppear:(BOOL)animated {
-    } else {
+        [dm setProjectID:curProjID];
+        } else {
         
```
This `} else {` block is indented one too many tabs inwards.

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.m:
```diff
@@ -31,27 +31,24 @@ - (void)viewWillAppear:(BOOL)animated {
         // Present an error message to the user and set project back to default project.
-        
         [self.view makeWaffle:@"Error: Selected Project did not have properly formatted fields. " duration:WAFFLE_LENGTH_LONG position:WAFFLE_BOTTOM image:WAFFLE_RED_X];
         int curProjID = 876;
```
This project ID shouldn't be hard-coded.

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.m:
```diff
@@ -31,27 +31,24 @@ - (void)viewWillAppear:(BOOL)animated {
         [self.view makeWaffle:@"Error: Selected Project did not have properly formatted fields. " duration:WAFFLE_LENGTH_LONG position:WAFFLE_BOTTOM image:WAFFLE_RED_X];
         int curProjID = 876;
         [dm setProjectID:curProjID];
         NSString *curProjIDStr = (curProjID > 100) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
```
Same idea with `curProjID > 100` - shouldn't it be `> 0`?

stowellm commented on Apps/DiceRoller/DiceRoller/ISDRProjectManager.m:
```diff
@@ -154,7 +150,6 @@ - (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)butto
                 [dm setProjectID:curProjID];
```
Again, do not hard-code the project ID (line 149).

stowellm commented:
At some point we'll need a `1024x1024` image of your icon for publishing to the app store.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.h:
```diff
@@ -10,15 +10,39 @@
+    //API, DataManager and DiceDataController
     API *api;
-    DiceDataController *diceController;
+    //DiceDataController *diceController;
```
Remove commented line.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -7,22 +7,44 @@
+    // if using the development iSENSE site, turn on development mode
+    [api useDev:true];
+    [nameBtn setTitle:[NSString stringWithFormat:@"Data Set Name: %@", dataSetName] forState:UIControlStateNormal];
     int projectID = 876;
```
No hardcoded project ID.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+}
+
+#pragma mark - Credentials
 
```
Since you don't have any other `#pragma` in your code, you can remove this one.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+#pragma end - Credentials
+
+#pragma mark - Name
+
```
Same for these other `#pragma`s.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+#pragma end - Name
+
+-(void) uploadDatadie1:(int)num1 die2: (int)num2 sumOfDies: (int)sumNum numOfTests: (int)numTest{
+    
```
Can we adjust the spacing of this? i.e. Remove the spaces between the `:` and `(int)`s:
`-(void) uploadDatadie1:(int)num1 die2:(int)num2 sumOfDies:(int)sumNum numOfTests:(int)numTest`

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+    dispatch_async(queue, ^{
+        
+        // normally, data would be passed into a method that makes this upload call.
+        // we will declare an arbitrary dice-roll and name for the data set
```
This comment is wrong.  We are not arbitrarily declaring dice values anymore.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+        // this example assumes the project has the fields "white dice", "yellow dice", and "sum"
+        
+        //NSString *userEmail = @"mobile.fake@example.com";
+        //NSString *password = @"mobile";
```
Delete this commented code and the comment above it.

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+        
+        // at this point, feel free to add more data points to the rowMajorData array!
+        // to prevent the need for parsing the project's fields and logging in multiple times,
+        // make sure you only make these API calls once though.
```
This set of comments can be deleted:
```
+        // at this point, feel free to add more data points to the rowMajorData array!
+        // to prevent the need for parsing the project's fields and logging in multiple times,
+        // make sure you only make these API calls once though.
```

stowellm commented on Apps/DiceRoller/DiceRoller/ViewController.m:
```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+        // you're done! this uploadDataToProject method returns a data set ID number that you can store or check.
+        // this dataSetID variable will now contain either a positive integer if the data uploaded
+        // successfully (namely the ID of your new data set), or -1 if the upload failed.  if your upload failed,
+        // you may have misformatted the data
```
At this point we may want some error checking, or even some basic user feedback.  For example, if the data set ID > 0, then the upload was likely successful, and we can show a brief waffle success message.  Otherwise display a waffle error message saying that the upload was unsuccessful.

stowellm commented on Apps/DiceRoller/ISDRTutorialViewController.m:
```diff
@@ -0,0 +1,131 @@
+#import "ISDRTutorialViewController.h"
+#import "ViewController.h"
+@interface ISDRTutorialViewController ()
+@end
```
Put a space between the `#import` and the `@interface`.

stowellm commented on Apps/DiceRoller/ISDRTutorialViewController.m:
```diff
@@ -0,0 +1,131 @@
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
```
If we don't need this commented out method, it can be removed.

stowellm commented on Apps/DiceRoller/PageContentViewController.h:
```diff
@@ -0,0 +1,19 @@
+#import <UIKit/UIKit.h>
+
+@interface PageContentViewController : UIViewController
+@property (weak, nonatomic) IBOutlet UIImageView *backgroundImg;
```
Space between the `@interface` and `@property` lines.

stowellm commented on Apps/DiceRoller/PageContentViewController.m:
```diff
@@ -0,0 +1,40 @@
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
```
Do we need this commented method?

stowellm commented:
Okay @Rajia, this is reviewed but needs quite a bit of tweaking as per my comments above before we merge this pull request.  If you have any questions about any of them, just let me know.

Rajia committed:
Made all the changes Mike requested in the pull request comments

Rajia committed:
Added waffle to notify user if upload was successful.

Rajia commented:
Thanks for reviewing @stowellm . I made all the changes you recommended and reassigned it to you to review once again. Please let me know if I missed anything! 

Also, I added waffles like you asked, however they're not showing up on the simulator. I'm going to take a look at this on Monday when I'm actually in the lab. I think they're actually working but I think they're just not showing up on the simulator. 

 Also, I'll get you the 1024 x 1024 icon on Monday.

Rajia committed:
Added 1024 app icon

stowellm commented on Apps/PageViewDemo:
```diff
@@ -0,0 +1 @@
+Subproject commit 89c0dbe047fecfb4110dbe15b91a4ba64d538b1d
```
What is this PageViewDemo file?  It can probably be removed.

stowellm commented:
Hi @Rajia - just the 3 comments above to be addressed then you're good to pull!  Nice work :smile:

Rajia committed:
Removed PageViewDemo and made changes in comments in constants file

Rajia commented:
@stowellm I removed the pageViewFile and moved the dialog constant to be under the correct comment. Fred said that "My Data Set" is fine but Jim wants us to ask the user for their name. Should we do that or should we leave it constant with motion? Either way, I think it's all-set to be merged in! If you would like me to add a dialog box when the user first opens the app for them to enter there name, let me know and I will do so!

stowellm commented:
Everything is good! I will merge it.
